library/ssl_tls13_keys.c

```diff
@@ -150,7 +148,7 @@ int mbedtls_ssl_tls13_hkdf_expand_label(
     psa_key_derivation_operation_t operation =
         PSA_KEY_DERIVATION_OPERATION_INIT;
 
-    if (label_len > MBEDTLS_SSL_TLS1_3_KEY_SCHEDULE_MAX_LABEL_LEN) {
+    if (label_len > MBEDTLS_SSL_TLS1_3_HKDF_LABEL_MAX_LABEL_LEN) {
         /* Should never happen since this is an internal
```

Comment:
Isn't this still an internal error? The function isn't directly called by the library user and we're checking the length before calling it.

Reply:
Oops sorry I missed the other check on my re-review as I was looking for it in `mbedtls_ssl_tls13_key_exporter()`. Yes this is still an internal error for the label length, so that part is okay as it is.

I believe it is still possible to trigger an internal error by requesting too much keying material though, unless I have missed something there as well?

Reply:
No, there's a check for that in `mbedtls_ssl_export_keying_material`.

```
    if (key_len > MBEDTLS_SSL_EXPORT_MAX_KEY_LEN) {
        return MBEDTLS_ERR_SSL_BAD_INPUT_DATA;
    }
```

I have to admit, it's a bit messy because some limits are different for TLS 1.2 and 1.3. But for the exported key length, I've decided to use one global limit that works in all cases because that can even depend on the ciphersuite, and I don't want to deal with that.

Reply:
Ah that's a good point, I had missed that. However, I would like to have a close look to see if this is correct. I'm unsure whether this is a sufficient lower bound, for example in TLS 1.2 where possibly a shorter hash function may be used. I will take a look and get back to you.

Reply:
TLS 1.2 uses a different PRF that *technically* can produce unlimited output but #9744 exists, so when using PSA there might indeed be a problem.

Reply:
I think happening across an existing underestimating bug is okay. The problem would be if we tried to generate more than the underlying PRF allows.